README.md

```diff
@@ -1,7 +1,488 @@
+  ...
+```
+### List of provided environment variables by SageMaker Containers
+
+The list of the environment variables is logged and available in cloudwatch logs. From the example above:
+```bash
+SM_NUM_GPUS=1
+SM_NUM_CPUS=4
+SM_NETWORK_INTERFACE_NAME=ethwe
+
+SM_CURRENT_HOST=algo-1
+SM_HOSTS=["algo-1","algo-2"]
+SM_LOG_LEVEL=20
+
+SM_USER_ARGS=["--batch-size","256","--learning-rate","0.0001","--communicator","pure_nccl"]
+
+SM_HP_LEARNING_RATE=0.0001
+SM_HP_BATCH-SIZE=10000
+
+SM_HPS={"batch-size": '256', "learning-rate": "0.0001","communicator": "pure_nccl"}
+
+SM_CHANNELS=["testing","training"]
+SM_CHANNEL_TRAINING=/opt/ml/input/data/training
+SM_CHANNEL_TESTING=/opt/ml/input/data/test
+
+SM_MODULE_NAME=user_script
+SM_MODULE_DIR=s3://sagemaker-{aws-region}-{aws-id}/{training-job-name}/source/sourcedir.tar.gz
+
+SM_INPUT_DIR=/opt/ml/input
+SM_INPUT_CONFIG_DIR=/opt/ml/input/config
+SM_OUTPUT_DIR=/opt/ml/output
+SM_OUTPUT_DATA_DIR=/opt/ml/output/data/algo-1
+SM_MODEL_DIR=/opt/ml/model
+
+SM_RESOURCE_CONFIG=
+{
+    "current_host": "algo-1",
+    "hosts": [
+        "algo-1",
+        "algo-2"
+    ]
+}
+
+SM_INPUT_DATA_CONFIG=
+{
+    "test": {
+        "RecordWrapperType": "None",
+        "S3DistributionType": "FullyReplicated",
+        "TrainingInputMode": "File"
+    },
+    "train": {
+        "RecordWrapperType": "None",
+        "S3DistributionType": "FullyReplicated",
+        "TrainingInputMode": "File"
+    }
+}
+
+
+SM_FRAMEWORK_MODULE=sagemaker_chainer_container.training:main
+
+SM_TRAINING_ENV=
+{
+    "channel_input_dirs": {
+        "test": "/opt/ml/input/data/testing",
+        "train": "/opt/ml/input/data/training"
+    },
+    "current_host": "algo-1",
+    "framework_module": "sagemaker_chainer_container.training:main",
+    "hosts": [
+        "algo-1",
+        "algo-2"
+    ],
+    "hyperparameters": {
+        "batch-size": 10000,
+        "epochs": 1
+    },
+    "input_config_dir": "/opt/ml/input/config",
+    "input_data_config": {
+        "test": {
+            "RecordWrapperType": "None",
+            "S3DistributionType": "FullyReplicated",
+            "TrainingInputMode": "File"
+        },
+        "train": {
+            "RecordWrapperType": "None",
+            "S3DistributionType": "FullyReplicated",
+            "TrainingInputMode": "File"
+        }
+    },
+    "input_dir": "/opt/ml/input",
+    "job_name": "preprod-chainer-2018-05-31-06-27-15-511",
+    "log_level": 20,
+    "model_dir": "/opt/ml/model",
+    "module_dir": "s3://sagemaker-{aws-region}-{aws-id}/{training-job-name}/source/sourcedir.tar.gz",
+    "module_name": "user_script",
+    "network_interface_name": "ethwe",
+    "num_cpus": 4,
+    "num_gpus": 1,
+    "output_data_dir": "/opt/ml/output/data/algo-1",
+    "output_dir": "/opt/ml/output",
+    "resource_config": {
+        "current_host": "algo-1",
+        "hosts": [
+            "algo-1",
+            "algo-2"
+        ]
+    }
+}
+```
+## IMPORTANT ENVIRONMENT VARIABLES
```

Comment:
In hindsight I think it's better not to have this important / unimportant distinction, but just to keep them ordered by how likely it is a user would need them.